backend/backend/container.py

```diff
@@ -111,32 +107,80 @@ def status_container(self, id: int) -> Status:
                 return Status.RESTARTING
             elif sts == "paused":
                 return Status.PAUSED
-            else:
-                return Status.UNDEFINED
+        return Status.UNDEFINED
 
-    def create_shared_folder(self, id: int, user: str):
-        """Creates a directory for the docker container with the specified id,
-        exposes the specified SSH port, and has SSH login credentials user/password.
-        Inside it creates a script for initialization that uses the specified username
-        for the home directory.
+    def create_shared_folder(self, container_id: str, user: str):
+        """Creates a directory for the docker container with the specified container_id,
+           exposes the specified SSH port, and has SSH login credentials user/password.
+           Inside it creates a script for initialization that uses the specified username
+           for the home directory
 
-        :param id: ID (name) of container
+        :param container_id: ID (name) of container
         :param port: SSH port that is exposed
         :param user: Username for SSH
         :param password: Password for SSH
         """
         # Save current working directory, must be able to restore later
-        current_dir = os.getcwd()
+        current_dir = self.root_path()
 
-        # Makes a shared folder named with the id
-        os.mkdir(str(id))
+        # Makes a shared folder named with the container_id
+        os.mkdir(os.path.join(current_dir, container_id))
 
         # Paths to files needed by the container
-        src = current_dir + "/template/"
-        dst = current_dir + "/" + str(id)
-        config_file = current_dir + "/" + \
-            str(id) + "/config/custom-cont-init.d/init.sh"
+        src = "template"
+        dst = current_dir + "/" + container_id
+        config_file = current_dir + "/" + container_id + "/config/custom-cont-init.d/init.sh"
```

Comment:
This could be done by copying into the folder for the volume that Docker has created. Would need to fetch volume path from Docker API.